src/relations/db.py

```diff
@@ -430,7 +393,61 @@ def _on_relation_broken(self, broken_event: RelationBrokenEvent):
 
         cfg.remove_user(user)
         self.charm.render_pgb_config(cfg, reload_pgbouncer=True)
-        self.charm.backend.postgres.delete_user(user)
+        if self.charm.unit.is_leader():
+            self.charm.peers.remove_user(user)
+            self.charm.backend.postgres.delete_user(user)
+
+    def update_databags(self, relation, updates: Dict[str, str]):
+        """Updates databag with the given dict."""
+        # Databag entries can only be strings
+        for key, item in updates.items():
+            updates[key] = str(item)
+
+        for databag in self.get_databags(relation):
+            databag.update(updates)
+
+    def _generate_username(self, event):
+        """Generates a unique username for this relation."""
+        app_name = self.charm.app.name
+        relation_id = event.relation.id
+        model_name = self.model.name
+        return f"{app_name}_user_id_{relation_id}_{model_name}".replace("-", "_")
```

Comment:
We can try to remove some things like the `id` text from this pattern like commented [here](https://github.com/canonical/postgresql-operator/pull/15#discussion_r940469355). I am still thinking if we can improve it a little bit more (but still thinking in cross model relations).